### DISCUSSION TITLE
Show or hide actions based on the focused application

### DISCUSSION THREAD

**Discussion by naikrovek** [elgatosf/elgato#90] [Ideas]
The "Profiles" feature does this, technically, but makes it very difficult to set up and maintain the behavior that I am after.

I want to be able to set each action shown under a button to be hidden or be shown based on the focused application.  If I want to hide a single button I have to create and maintain two profiles, and duplicate changes that I make to one profile in the other profile, and that _quickly_ gets exhausting.  I have buttons which I want to remain the same no matter what profile is active, and that means I must manually copy changes for those buttons to all profiles whenever I change the button in one of the profiles.

A UX concern arises, because you could have one physical button set to multiple actions depending on the active application, and I don't know how you'd approach this in the UI, but surely there is a way.  

So, rather than creating a dozen profiles which I must manually keep up to date every time I change something on a button that I want on all profiles, instead allow me to show/hide buttons based on the focused application.  For me, that would be a much better solution, and the current profile-based solution is almost completely useless.

**Comment by GeekyEggo** [comment]
Hi @naikrovek, this looks like a Stream Deck application feature request that isn’t related to the SDK. Feature requests relating to Stream Deck can be submitted on [our official feature request submission](https://e.lga.to/FeatureRequest) page; this will ensure it reaches the right people.
